### PR TITLE
Add ability to pass Faraday config block to Client

### PR DESCRIPTION
**Pull request by alexrudall**
- Add test that error messages are included in Faraday response body
- Add ability to pass Faraday config block to client

## All Submissions:

* [ ] Have you followed the guidelines in our [Contributing document](../blob/main/CONTRIBUTING.md)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?

### CODE REVIEW

File: spec/openai/client/chat_spec.rb [hunk 1]
```diff
@@ -24,6 +26,46 @@
...
+
+          it "raises an error containing the reason" do
+            VCR.use_cassette(cassette) do
+              response
+            rescue Faraday::Error => e
+              expect(e.response.dig(:body, "error",
+                                    "message")).to include("Missing parameter 'name'")
+            end
+          end
+        end
```
Comment (alexrudall) [inline spec/openai/client/chat_spec.rb:67]: This spec is kind of unrelated but being unable to find this error is what triggered this PR. So including it

**Comment by ErikDeBruijn** [comment]
This is such interesting timing. I was thinking of adding something along these lines, actually, because so far I was monkey patching this in the initializer `openai.rb`!

```
OpenAI.configure do |config|
  config.access_token = Rails.application.credentials.open_ai_key
  config.request_timeout = 300
end

module OpenAI
  module HTTP
    def connection_with_timeout(multipart: false)
      original_connection(multipart:).tap do |conn|
        conn.use TimeoutMiddleware, timeout: @request_timeout
      end
    end

    alias_method :original_connection, :conn
    alias_method :conn, :connection_with_timeout
  end
end
```

Obviously, having such a block to configure the Faraday object is an improvement.

FYI: this was because I was still observing time outs

**Comment by alexrudall** [comment]
> This is such interesting timing. I was thinking of adding something along these lines, actually, because so far I was monkey patching this in the initializer `openai.rb`!
> 
> 
> 
> ```
> 
> OpenAI.configure do |config|
> 
>   config.access_token = Rails.application.credentials.open_ai_key
> 
>   config.request_timeout = 300
> 
> end
> 
> 
> 
> module OpenAI
> 
>   module HTTP
> 
>     def connection_with_timeout(multipart: false)
> 
>       original_connection(multipart:).tap do |conn|
> 
>         conn.use TimeoutMiddleware, timeout: @request_timeout
> 
>       end
> 
>     end
> 
> 
> 
>     alias_method :original_connection, :conn
> 
>     alias_method :conn, :connection_with_timeout
> 
>   end
> 
> end
> 
> ```
> 
> 
> 
> Obviously, having such a block to configure the Faraday object is an improvement.
> 
> 
> 
> FYI: this was because I was still observing time outs

Ah sweet - could you test this PR with that use case? Would be much appreciated 🙏

**Review by obie — APPROVED** [review]
Please do merge

**Pull request merged**